### ISSUE TITLE
Why is `ort-server.params.kts` unconditionally read by a config file provider?

### ISSUE THREAD

**Issue by Etsija** [eclipse-apoapsis/ort-server#533]
When using GitHub config file provider, the context is resolved properly, after which ORT Server tries to unconditionally access `ort-server.params.kts` from the configuration repository. As our repository has no such file, this probably leads to the `Unexpected result of Config Worker: Ignored` error seen in the logs below.

I provided this issue with a `bug` tag, because the way I see it, either it is indeed a bug, or then it's an undocumented requirement of a configuration provider.

NOTE: this is not necessarily at all a bug in GitHub config file provider, maybe it's some other component wanting unconditional access to the file mentioned.

```
ort-server-config-worker-1     | 2024-06-28 06:48:56.342 [pool-1-thread-4] level=DEBUG o.e.a.o.c.g.GitHubConfigFileProvider - GitHub default branch: 'main'.
ort-server-config-worker-1     | 2024-06-28 06:48:56.381 [pool-1-thread-4] level=DEBUG o.e.a.o.c.g.GitHubConfigFileProvider - GET 'https://api.github.com/repos/doubleopen-project/ort-config/branches/main'
ort-server-config-worker-1     | 2024-06-28 06:48:57.090 [pool-1-thread-4] level=INFO  o.e.a.o.workers.config.ConfigWorker - Provided configuration context 'main' was resolved to '05af00f73554c579175745c1da30eaf53c50d3ff'.
ort-server-config-worker-1     | 2024-06-28 06:48:57.091 [pool-1-thread-4] level=DEBUG o.e.a.o.c.g.GitHubConfigFileProvider - GET 'https://api.github.com/repos/doubleopen-project/ort-config/contents/ort-server.params.kts?ref=05af00f73554c579175745c1da30eaf53c50d3ff'
...
ort-server-config-worker-1     | 2024-06-28 06:48:57.346 [pool-1-thread-4] level=ERROR o.e.a.o.w.config.ConfigComponent - Unexpected result of Config worker: Ignored
ort-server-config-worker-1     | 2024-06-28 06:48:57.351 [pool-1-thread-4] level=INFO  o.e.a.o.transport.MessagePublisher - Sending 'ConfigWorkerError' message to 'OrchestratorEndpoint'. TraceID: 'ca951239-4af7-4878-9c48-6f6aad178555'.
```

**Comment by mnonnenmacher** [comment]
If no request modification or verification is required it should not be required that the script exists.

**Comment by Etsija** [comment]
Can you elaborate: request modification or verification? Related to #532, GitHub config provider returns so little information about the files it reads / cannot find / auth failures, that I had no other idea in the above run but deduce from the logs that it is expected that the provider has the script available, and when it doesn't find it, it throws that mystical `Unexpected result of Config worker: Ignored` exception.

tl;dr from the logs it is not possible to find out what went wrong.

**Comment by Etsija** [comment]
OK so I may be completely wrong, but I don't think this works as intended: https://github.com/eclipse-apoapsis/ort-server/blob/63cd0fe7908d5b5110af95520b3a6e892acbafd5/workers/config/src/main/kotlin/ConfigWorker.kt#L85-L113

Because when the validation script doesn't exist (like in our case), the result is always `RunResult.Ignored`. Then the `endPointHandler` always throws an error:
https://github.com/eclipse-apoapsis/ort-server/blob/63cd0fe7908d5b5110af95520b3a6e892acbafd5/workers/config/src/main/kotlin/ConfigComponent.kt#L52-L67

**Comment by Etsija** [comment]
```
is RunResult.Ignored -> { 
    logger.info("Config worker job succeeded for run '$runId', no validation script used.") 
    ConfigWorkerResult(runId) 
} 
```

**Comment by mnonnenmacher** [comment]
@Etsija Please check if #538 fixes the issue for you.

**Comment by Etsija** [comment]
Looking at the changes in that PR, I think it would at least let the worker succeed and the run proceed further. I'll retest when #538 is merged and close this issue correspondingly.
